### DISCUSSION TITLE
why does ./main use vram even if i did -ngl 0 -ngld 0?

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3587] [Q&A] [answered]
how to make it only use cpu ram?

i was running 2 instance.
1 in pure vram (e.g. -ngl 35)
1 in cpu (which uses vram)
and i cant run another instance. would like to ran all instances

anyone knows how to limit to running in cpu ram only?

**Comment by KerfuffleV2** [comment, marked as answer]
Prompt processing can use VRAM even when not offloading layers. As far as I know, there isn't a way to disable that, so basically the solution would be to compile an executable that only has CPU support and use that in the case you want to be use it won't touch the GPU.

Just note that ingesting the prompt/running stuff like `perplexity` is much, much slower on CPU.

**Reply by hiqsociety** [reply]
@KerfuffleV2 make sense to compile another executable. it's strange but thx. i'm doing so now. if it works, will close this discussion.

**Reply by KerfuffleV2** [reply]
I'll be pretty surprised if it uses VRAM when compiled without GPU support. :)

**Reply by hiqsociety** [reply]
@KerfuffleV2 you are a genius! another pure cpu compiled instance work.

by the way, possible to fix this? :D

much prefer to keep 1 instance instead.

**Reply by hiqsociety** [reply]
@KerfuffleV2 on second thoughts, i can live with this. 
u guys have better things to do.
i hope u guys can do awq file running? :D that'll be super awesome.
